runtime/shared_common/ClasspathManagerImpl2.cpp

```diff
@@ -629,7 +629,7 @@ SH_ClasspathManagerImpl2::cpeTableUpdate(J9VMThread* currentThread, const char*
 	addToList = cpeTableLookupHelper(currentThread, name, nameLen, isToken);
 
 	if (addToList==NULL) {
-		returnVal = cpeTableAdd(currentThread, name, nameLen, CPEIndex, item, isToken, doTag, cachelet);
+		returnVal = cpeTableAdd(currentThread, name, nameLen, CPEIndex, item, isToken, flags, doTag, cachelet);
 	} else {
```

Comment:
I think the flags of the existing entry need to be updated when the entry already exists.

Reply:
As confirmed with @pshipton,  there is no need to set up the flags via storeNew() as they are already set up via notifyClasspathChange3() in all places. So I will remove all related code there (under the way).

Reply:
I have a little bit worried about the jars on the bootstrap class path. The change in SharedClassURLClasspathHelperImpl.java won't have effect on bootstrap class path. But you can verified the behavior by enabling some trace points to see how many times we check for the timestamp of on bootstrap jars.

Reply:
I already double-checked the traces which indicates the existing implementation already works on the the bootstrap class path via `notifyClasspathEntryStateChange`.  The new changes here only affects the extension class path e.g. 
```
C:\>jdk8_openj9\bin\java  -Xtrace:iprint={j9shr.366,j9jcl.481,j9jcl.485,j9shr.1199}  -Xshareclasses:name=spring,noTimestampChecks   -version
...
> JCL: SharedClassURLClasspathHelperImpl notifyClasspathChange3: Entering
> CPM notifyClasspathEntryStateChange: Entering with path=C:\\jdk8_openj9\jre\lib\ext\access-bridge-64.jar, newState=0x1
> CPM notifyClasspathEntryStateChange: Entering with path=C:\\jdk8_openj9\jre\lib\ext\cldrdata.jar, newState=0x1
> CPM notifyClasspathEntryStateChange: Entering with path=C:\\jdk8_openj9\jre\lib\ext\dnsns.jar, newState=0x1
> CPM notifyClasspathEntryStateChange: Entering with path=C:\\jdk8_openj9\jre\lib\ext\dtfj.jar, newState=0x1
> CPM notifyClasspathEntryStateChange: Entering with path=C:\\jdk8_openj9\jre\lib\ext\dtfjview.jar, newState=0x1
> CPM notifyClasspathEntryStateChange: Entering with path=C:\\jdk8_openj9\jre\lib\ext\jaccess.jar, newState=0x1
> CPM notifyClasspathEntryStateChange: Entering with path=C:\\jdk8_openj9\jre\lib\ext\localedata.jar, newState=0x1
> CPM notifyClasspathEntryStateChange: Entering with path=C:\\jdk8_openj9\jre\lib\ext\nashorn.jar, newState=0x1
> CPM notifyClasspathEntryStateChange: Entering with path=C:\\jdk8_openj9\jre\lib\ext\sunec.jar, newState=0x1
> CPM notifyClasspathEntryStateChange: Entering with path=C:\\jdk8_openj9\jre\lib\ext\sunjce_provider.jar, newState=0x1
> CPM notifyClasspathEntryStateChange: Entering with path=C:\\jdk8_openj9\jre\lib\ext\sunmscapi.jar, newState=0x1
> CPM notifyClasspathEntryStateChange: Entering with path=C:\\jdk8_openj9\jre\lib\ext\sunpkcs11.jar, newState=0x1
> CPM notifyClasspathEntryStateChange: Entering with path=C:\\jdk8_openj9\jre\lib\ext\traceformat.jar, newState=0x1
> CPM notifyClasspathEntryStateChange: Entering with path=C:\\jdk8_openj9\jre\lib\ext\zipfs.jar, newState=0x1
< JCL: SharedClassURLClasspathHelperImpl notifyClasspathChange3: Exiting
> JCL: SharedClassURLClasspathHelperImpl notifyClasspathChange3: Entering
< JCL: SharedClassURLClasspathHelperImpl notifyClasspathChange3: Exiting
openjdk version "1.8.0_222-internal"
OpenJDK Runtime Environment (build 1.8.0_222-internal-administrator_2019_06_21_20_50-b00)
Eclipse OpenJ9 VM (build sc_no_classpath_timestamp_check-0bff4c4f8, JRE 1.8.0 Windows 10 amd64-64-Bit Compressed References 20190621_000000 (JIT enabled, AOT enabled)
OpenJ9   - 0bff4c4f8
OMR      - 7c874228
JCL      - 6cc7e03657 based on jdk8u222-b05)
```

Reply:
> I already double-checked the traces which indicate the existing implementation already works on the the bootstrap class path

So the zip_hook is still working for bootstrap class path. After this change, we are still using the zip_hook for the bootstrap class path. And the new change here only affects non-bootstrap class path. If this is the desired behavior, we might want to mention in the documentation that new options "checkClasspathTimestamp" and "noCheckClasspathTimestamp" affects non-bootstrap class path only.